src/app/models/profile.ts

```diff
@@ -6,8 +6,11 @@ export class Profile {
     bio?: string;
     username?: string;
     url?: string;
-    emails?: Email[];
-    primaryEmail?: Email;
+    emails?: string[];
```

Comment:
I was thinking that if you want to change the type or add a type then you need to change the code. But if you store the type then you can add it to the server side and just fetch it. The data model should still reflect the UX, just not in a hard-coded fashion.